### PR TITLE
refine-ZK-3871

### PR DESCRIPTION
**Pull request by klyvechen**

### CODE REVIEW

**Comment by scribetw** [comment]
Please merge this before https://github.com/zkoss/zk/pull/1179

File: zk/src/org/zkoss/zk/ui/sys/ShadowElementsCtrl.java [hunk 1]
```diff
@@ -128,12 +128,13 @@ private static void doBindChildrenInFilter(Component[] shadows) {
 				if (!se.getDistributedChildren().isEmpty() || !((ShadowElementCtrl) se).isDynamicValue())
 					continue;
 				List<Component> children = shadow.getChildren();
-				if (children.size() > 0) {
-					doBindChildrenInFilter(children.toArray(new Component[children.size()]));
-				} else {
+				if (shadow.getChildren().size() == 0) {
```
Comment (DevChu) [inline zk/src/org/zkoss/zk/ui/sys/ShadowElementsCtrl.java]: duplicate to List<Component> children = shadow.getChildren();

Comment (klyvechen) [reply zk/src/org/zkoss/zk/ui/sys/ShadowElementsCtrl.java]: The status the shadow.getChildren() may change.

**Comment by scribetw** [comment]
@zkadmin rebase

**Comment by zkadmin** [comment]
I just pushed up the changes, enjoy!

**Pull request merged**